analyses/snv-callers/scripts/02-merge_callers.R

```diff
@@ -67,6 +67,69 @@ option_list <- list(
 # Parse options
 opt <- parse_args(OptionParser(option_list = option_list))
 
+vaf_filter <- opt$vaf_filter # get out of opt list for sql
+
+############################## Custom Functions ################################
+
+#' Split multinucleotide variants into single nucleotide calls
+#'
+#' @param mnv_tbl a table containing MNVs (may be from an sql connection)
+#'
+#' @return a data frame of SNVs derived from the MNVs
+#'   (does not contain any of the original SNVs)
+split_mnv <- function(mnv_tbl) {
+  mnv_df <- mnv_tbl %>%
+    dplyr::filter(Variant_Type %in% c("DNP", "TNP", "ONP")) %>%
+    as.data.frame() %>%
+    # temp_id is not currently used, but will be handy for reconstituting MNVs
+    dplyr::mutate(temp_id = dplyr::row_number()) %>%
+    # lancet adds a base to the start of MNV Allele Calls
+    dplyr::mutate(
+      Allele = dplyr::case_when(
+        stringr::str_length(Allele) == stringr::str_length(Reference_Allele) ~
+          Allele,
+        stringr::str_length(Allele) == stringr::str_length(Reference_Allele) + 1 ~
+          stringr::str_sub(Allele, 2)
+      )
+    )
+  # separate into individual rows
+  if ("Match_Norm_Seq_Allele1" %in% colnames(mnv_df)) {
```

Comment:
What's the rationale of using Match_Norm_Seq_Allele if it exists?

Reply:
I'm not using it, just preserving it if it exists. Otherwise every site would still have the multibase calls in this column. Added a more detailed comment